datalad/support/gitrepo.py

```diff
@@ -2837,7 +2838,7 @@ def add_submodule(self, path, name=None, url=None, branch=None):
             cmd += ['-b', branch]
         if url is None:
             # repo must already exist locally
-            subm = GitRepo(op.join(self.path, path), create=False, init=False)
+            subm = repo_from_path(op.join(self.path, path))
```

Comment:
Stepping through this with the added tests and `eval_under_testloopfs`, this looks good to me.